### PR TITLE
feat: remove 404 pages from the sitemap. closes #113

### LINKED ISSUE #113: 404.html appearing in sitemap.xml
I have a 404.html file in my root. When I view the contents of sitemap.xml, I see:

```
...
<url>
<loc>https://example.com/404.html</loc>
</url>
...
```

Should this page be excluded from the sitemap?

### PR DESCRIPTION
**Pull request by jhabdas**
poof. fixes #113.

### CODE REVIEW

**Comment by jhabdas** [comment]
> Could also be a Document, in addition to a static file, right?

Sorry, something's up with GitHub and I'm not able to respond directly to your review comment. Do you mean `404.md`? If so, all markdown files are converted to HTML when output in the sitemap and testing that conversion is outside the scope of this plug-in or test.

**Comment by jhabdas** [comment]
I do what I can.

**Comment by pathawks** [comment]
@jhabdas Sorry, I was confused.

What I meant is that `404.html` could be a static file (without front matter), so we should probably also check in the third loop.

It is probably unlikely that somebody is outputting `404.html` from a collection. (Somebody will quote this line in an issue a year from now, I am sure. 😛)

**Comment by jhabdas** [comment]
Pareto principal say press that merge button. I'm off to bed.

**Comment by benbalter** [comment]
I believe ` {% for file in page.static_files %}` needs the same logic, in case the 404 file doesn't have YAML front matter.

**Comment by jhabdas** [comment]
@benbalter oic. on it

**Comment by jhabdas** [comment]
@pathawks @benbalter thanks for ensuring excellence in your work. I've added the missing fixture and test. you guys have eagle eyes, I swear.

File: lib/sitemap.xml [hunk 1]
```diff
@@ -26,7 +26,9 @@
     </url>
   {% endfor %}
 
-  {% for file in page.static_files %}
+  {% assign static_files = page.static_files | where_exp:'page','page.name != "404.html"' %}
+  {% for file in static_files %}
+    {{ page.path }}
```
Comment (pathawks) [inline lib/sitemap.xml]: We probably don't need to output the path here.

**Comment by pathawks** [comment]
One comment. Otherwise LGTM

Thanks for getting this done :beers:🎉

File: lib/sitemap.xml (hunk 1 above)
Comment (jhabdas) [inline lib/sitemap.xml]: oops

**Comment by jhabdas** [comment]
@pathawks all set with a02af55

File: spec/jekyll-sitemap_spec.rb [hunk 2]
```diff
@@ -93,10 +93,14 @@
     expect(contents).to match /\/some-subfolder\/htm\.htm/
   end
 
-  it "does include assets or any static files with .pdf extension" do
+  it "does include assets any static files with .pdf extension" do
```
Comment (pathawks) [inline spec/jekyll-sitemap_spec.rb]: Why was this changed?

Comment (jhabdas) [inline spec/jekyll-sitemap_spec.rb]: I'm sorry, @pathawks restoring

**Comment by pathawks** [comment]
@jekyllbot: merge +minor

**Comment by pathawks** [comment]
@jhabdas Thank you so much! I appreciate your patience.

**Comment by DirtyF** [comment]
No more need for `sitemap: false` in `404.html` 🎉
